### DISCUSSION TITLE
Recommend method for creating frozen modules to use with MicroPython?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16018] [Core Development]
Recommend method for creating frozen modules to use with MicroPython?  Been a while since I have done this, and I don't recall the explicit details, so trying to get back to this.  In this case I want to add the uPing module I have to my ESP32 as efficiently as possible, short of merging said module into a custom MicroPython image.

**Comment by glenn20** [comment]
Freezing a module does mean compiling it into the micropython image (see https://docs.micropython.org/en/latest/reference/manifest.html).

Perhaps, you are thinking of byte compiling your .py files into .mpy files (see https://docs.micropython.org/en/latest/reference/mpyfiles.html).

Thanks to the work of Andrew and Damien you can now `pip install mpy-cross` (https://pypi.org/project/mpy-cross/).

**Comment by Jibun-no-Kage** [comment]
Yes.... I was thinking of byte compiling in my original question.  But I am just compile a new image in the end... been a while since I have, but should not be too sticky to refresh an old skill set.  Thank you.
